sentry/src/main/java/io/sentry/SentryEnvelopeItem.java

```diff
@@ -151,6 +153,50 @@ public static SentryEnvelopeItem fromUserFeedback(
     return new SentryEnvelopeItem(itemHeader, () -> cachedItem.getBytes());
   }
 
+  public static SentryEnvelopeItem fromAttachment(
+      @NotNull ILogger logger, final @NotNull Attachment attachment) {
+
+    final CachedItem cachedItem =
+        new CachedItem(
+            () -> {
+              if (attachment.getBytes() != null) {
+                return attachment.getBytes();
+              } else if (attachment.getPathname() != null) {
+                try (FileInputStream fileInputStream =
+                        new FileInputStream(attachment.getPathname());
+                    BufferedInputStream inputStream = new BufferedInputStream(fileInputStream);
+                    ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+
+                  byte[] bytes = new byte[1024];
+                  int length;
+                  int offset = 0;
+                  while ((length = inputStream.read(bytes)) != -1) {
+                    outputStream.write(bytes, offset, length);
+                  }
+                  return outputStream.toByteArray();
+                } catch (IOException | SecurityException exception) {
+                  logger.log(
+                      SentryLevel.ERROR,
+                      exception,
+                      "Serializing attachment %s failed.",
+                      attachment.getPathname());
+                }
+              }
+
+              return new byte[0];
```

Comment:
This means we'll be sending an 0 byte attachments when we have errors reading?
Just want to make sure we've thought this through, as it will be confusing to customers given no additional data to why this was a 0 byte file.

Reply:
I changed it. Please have a look.